unified-scheduler-pool/src/lib.rs

```diff
@@ -1111,13 +1109,13 @@ impl<S: SpawnableScheduler<TH>, TH: TaskHandler> ThreadManager<S, TH> {
                                 .unwrap();
                             result_with_timings = new_result_with_timings;
                         }
-                        Ok(_) => {
-                            unreachable!();
-                        }
                         Err(_) => {
-                            // This unusual condition must be triggered by ThreadManager::drop();
+                            // This unusual condition must be triggered by ThreadManager::drop().
+                            // Initialize result_with_timings with a harmless value...
+                            result_with_timings = initialized_result_with_timings();
```

Comment:
just by adding this, it turned out we can please rust borrow checker avoiding use of `std::mem::replace()` as was before.